### ISSUE TITLE
addProgressionEvent doesn't send scores and is incorrectly typed

### ISSUE THREAD

**Issue by ubershmekel** [GameAnalytics/GA-SDK-JAVASCRIPT#220]
I'm using:

```
GameAnalytics.addProgressionEvent(EGAProgressionStatus.Complete, this.language, "level" + this.level, undefined, durationSeconds);
```

I see this in the log:

```
Info/GameAnalytics: Add PROGRESSION event: {status:Complete, progression01:japanese, progression02:level1, progression03:, score:7.255, attempt:1}
POST https://api.gameanalytics.com/v2/52e2232c1978d43ea0a92c8843b41cb5/events 400
Warning/GameAnalytics: Event queue: Failed to send events.
```

Looking at the response body:

```
[{"event":{"v":2,"user_id":"5f78d159-71c3-41d5-0a44-51f349f0e2d6","client_ts":1630957757,"sdk_version":"javascript 4.2.1","os_version":"windows 10.0","manufacturer":"unknown","device":"unknown","browser_version":"chrome 93","platform":"windows","session_id":"f53ac4c9-e5b6-4a75-417b-1f78e06b9e7e","session_num":7,"connection_type":"lan","category":"progression","event_id":"Complete:japanese:level1","score":7.255,"attempt_num":1},"errors":[{"error_type":"wrong_type","path":"/score"}]}]
```

So the `score` is the wrong type. But if I make it a string then it doesn't get sent at all. See https://github.com/GameAnalytics/GA-SDK-JAVASCRIPT/blob/aa89644ca8f3e08a14a1d8fe289e445183776369/src/GameAnalytics.ts#L344

Just now after writing up this issue, I thought of the idea to make the score number an integer. That seems to work. Why would the typescript use the `any` type for `score` then?

This is a pretty rigid data scheme you got there. I personally wanted to log two different scores (amount of mistakes and the time to completion). For my game at https://ubershmekel.github.io/nihongo-shooter/

Seems gameanalytics.com might not be a good fit.

**Comment by the1schwartz** [comment]
Thanks for reporting this. The events schema only accepts whole numbers so I have updated the code to round off float numbers to accept them in our backend. It is fixed in the latest version (v4.2.2)

**Comment by ubershmekel** [comment]
Thank you for https://github.com/GameAnalytics/GA-SDK-JAVASCRIPT/pull/221/files
